api/src/pcapi/core/offers/validation.py

```diff
@@ -504,6 +506,28 @@ def check_offer_subcategory_is_valid(offer_subcategory_id: str | None) -> None:
         raise exceptions.SubCategoryIsInactive()
 
 
+def check_product_ean_valide_for_record_store_venue(
+    offer_subcategory_id: str | None, ean_code: str | None, venue_type_code: VenueTypeCode
+) -> None:
+    if venue_type_code != VenueTypeCode.RECORD_STORE:
+        return
+    if offer_subcategory_id not in [
+        subcategories.SUPPORT_PHYSIQUE_MUSIQUE_CD.id,
+        subcategories.SUPPORT_PHYSIQUE_MUSIQUE_VINYLE.id,
+    ]:
+        return
```

Comment:
```suggestion
    if (venue_type_code != VenueTypeCode.RECORD_STORE) and (offer_subcategory_id not in [
        subcategories.SUPPORT_PHYSIQUE_MUSIQUE_CD.id,
        subcategories.SUPPORT_PHYSIQUE_MUSIQUE_VINYLE.id,
    ]):
        return
```

Reply:
En effet, je suis d'accord avec ta suggestion. J'avais fait comme ça pour être super explicite sur les conditions et n'avoir qu'à supprimer le `if` le jour où on utilise cette validation pour toutes les venues. Mais en effet, ça change pas grand chose de regrouper dans un seul `if`

Reply:
Pour plus de lisibilité, je vais laisser les 2 conditions séparées. Il s'agirait d'un `or` plutôt que d'un `and`.